jina/peapods/pods/__init__.py

```diff
@@ -114,10 +114,6 @@ def _parse_args(self, args: Namespace) -> Dict[str, Optional[Union[List[Namespac
                 self.is_tail_router = True
                 peas_args['tail'] = _copy_to_tail_args(args)
             peas_args['peas'] = _set_peas_args(args, peas_args.get('head', None), peas_args.get('tail', None))
-        else:
-            self.is_head_router = False
```

Comment:
I think this logic needs to remain there, are you sure it needs removal?

Reply:
thanks, I don't think it is used, but now I keep it there because it is out of scope for this pr